plugins/nodes/src/js/components/NodesTable.tsx

```diff
@@ -155,7 +163,7 @@ export default class NodesTable extends React.Component<
   componentWillReceiveProps(nextProps: NodesTableProps): void {
     this.setState(
       this.updateData(
-        nextProps.hosts.getItems(),
+        nextProps.hosts ? nextProps.hosts.getItems() : null,
```

Comment:
do we even need updating data when we don't get `nextProps.hosts`?

Reply:
sorting comes from state 🤔

Reply:
I can imagine some situations where we do want to update even when `nextProps.hosts` is null or undefined or empty.

Reply:
Ok, I see 1 case at least. When for some reason `props.hosts` turn to undefined we want to `nullify` the `state.data`  thus showing loader.